src/fns/constrained_ops.nr

```diff
@@ -235,9 +238,16 @@ pub(crate) fn validate_in_field<let N: u32, let MOD_BITS: u32>(
 **/
 pub(crate) fn validate_in_range<let N: u32, let MOD_BITS: u32>(limbs: [Field; N]) {
     for i in 0..(N - 1) {
-        limbs[i].assert_max_bit_size::<120>();
+        // this is a hack, the assert_max_bit_size is not working when limbs[i] is not a constant
+        // limbs[i].assert_max_bit_size::<120>();
+        assert(limbs[i].lt(2.pow_32(120)), "call to assert_max_bit_size 2");
     }
-    limbs[N - 1].assert_max_bit_size::<MOD_BITS - ((N - 1) * 120)>();
+    // this is a hack, the assert_max_bit_size is not working when limbs[i] is not a constant
+    // limbs[N - 1].assert_max_bit_size::<MOD_BITS - ((N - 1) * 120)>();
+    assert(
+        limbs[N - 1].lt(2.pow_32((MOD_BITS as Field) - ((N - 1) * 120) as Field)),
+        "call to assert_max_bit_size",
+    );
```

Comment:
The tests are all passing with the commented out range checks. Is this still an issue?

Reply:
@kashbrti

Reply:
I think the asserts that I've added are doing the range check if I'm not making a mistake. 
The issue was that as `limb[N-1]` is not a constant, the range check was happening for the bit size of the native field rather than the actual input.

Reply:
Yep, I agree that the added asserts are going a less efficient range check.

> The issue was that as limb[N-1] is not a constant, the range check was happening for the bit size of the native field rather than the actual input.

Why would a non-constant field result in us not performing a check whether `limb[N-1]` fits within `MOD_BITS - (N-1)*120` bits?

Reply:
Did you verify the output in the ACIR? If we're producing incorrect output here we shouldn't just hide that by rewriting the Noir.

Reply:
To be 100% clear, I'd like to restore the original calls to `assert_max_bit_size` so I'm keen to know why they were replaced and if it's due to a correctness issue in the compiler then we need to fix that.